storage/innobase/buf/buf0dblwr.cc

```diff
@@ -338,6 +337,92 @@ buf_dblwr_create(void)
 	goto start_again;
 }
 
+/** Compute the path to the parallel doublewrite buffer, if not already done */
+__attribute__((warn_unused_result))
+static
+dberr_t
+buf_parallel_dblwr_make_path(void)
+{
+	if (parallel_dblwr_buf.path)
+		return(DB_SUCCESS);
+
+	char path[FN_REFLEN];
+
+	if (srv_parallel_doublewrite_path
+	    && is_absolute_path(srv_parallel_doublewrite_path)) {
+
+		strncpy(path, srv_parallel_doublewrite_path, sizeof(path));
+	} else if (srv_data_home[strlen(srv_data_home) - 1]
```

Comment:
srv_data_home might be "" legitimately, see http://bugs.mysql.com/bug.php?id=78831